src/components/CodeManagement/index.jsx

```diff
@@ -0,0 +1,220 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import Helmet from 'react-helmet';
+import createHistory from 'history/createBrowserHistory';
+import { Icon, Pagination } from '@edx/paragon';
+
+import H2 from '../H2';
+import Hero from '../Hero';
+import Coupon from '../Coupon';
+import LoadingMessage from '../LoadingMessage';
+import StatusAlert from '../StatusAlert';
+
+class CodeManagement extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.couponRefs = [];
+
+    this.handleCouponExpand = this.handleCouponExpand.bind(this);
+  }
+
+  componentDidMount() {
+    const { enterpriseId, enterpriseSlug } = this.props;
+    const history = createHistory();
+
+    if (enterpriseId) {
+      this.props.fetchCouponOrders();
+    }
+
+    this.props.fetchPortalConfiguration(enterpriseSlug);
+
+    // Reset `hasRequestedCodes` so that status alert disappears
+    if (history.location.state && history.location.state.hasRequestedCodes) {
+      const state = { ...history.location.state };
+      delete state.hasRequestedCodes;
+      history.replace({ ...history.location, state });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { enterpriseId } = this.props;
+    if (enterpriseId && enterpriseId !== prevProps.enterpriseId) {
+      this.props.fetchCouponOrders();
```

Comment:
do you need to `fetchPortalConfiguration` here as well or is that covered up the stack?

Reply:
We should only need it in the initial `componentDidMount` as the `enterpriseSlug` will change up the stack in the prop that gets passed to the route for code management.

For the `Admin` page, we only call it in `componentDidMount` as well.